File: cp_rust/src/bin/abc398-b-full-house-3.rs
// Created by Ayush Biswas at 2026/07/08 16:06
// https://atcoder.jp/contests/adt_medium_20250605_3/tasks/abc398_b
use cp_lib::*;

// @code begin
use cpio::*;
use itertools::Itertools;

sol! {
    fn solution(
        cards: [usize]
    ) -> Bool {
        let groups = cards.into_iter().sorted().group_by(|&x| x).map(|g| g.len()).collect_vec();
        let twos = groups.iter().count_where(|&&x| x >= 2);
        let threes = groups.iter().count_where(|&&x| x >= 3);
        (twos >= 2 && threes > 0).into()
    }
}
// @code end

File: cp_nim/src/bin/abc398-b-full-house-3.nim
# Created by Ayush Biswas at 2026/07/19 09:32
# https://atcoder.jp/contests/adt_medium_20250605_3/tasks/abc398_b
import ../lib/cpio
import ../lib/itertools

# @code begin

import sugar, sequtils, tables, algorithm

var r = newReader()

let a = r[seq[int]].groupBy(k => k)

let sizes = a.values.toSeq.map(x => x.len).sorted

if sizes.len >= 2 and sizes[^1] >= 3 and sizes[^2] >= 2:
  echo "Yes"
else:
  echo "No"

# @code end
